File: ios/SoccerOdds2026/Models/OddsHistory.swift
import Foundation

/// Keyed by match ID; each value is a chronologically ordered array of daily snapshots.
typealias OddsHistory = [String: [OddsSnapshot]]

struct OddsSnapshot: Codable, Hashable {
    let t: String      // "YYYY-MM-DD"
    let home: Double
    let draw: Double
    let away: Double
}

File: pipeline/src/oddsHistory.js
// pipeline/src/oddsHistory.js
// Append one rounded daily 1X2 snapshot per match. Idempotent per (matchId, day):
// a second run on the same UTC day overwrites that day's snapshot.
// history shape: { [matchId]: [ { t: "YYYY-MM-DD", home, draw, away }, ... ] }
export function updateHistory(history, matches, isoNow) {
  const day = isoNow.slice(0, 10);
  const next = { ...history };
  for (const m of matches) {
    const snap = { t: day, home: m.probs_1x2.home, draw: m.probs_1x2.draw, away: m.probs_1x2.away };
    const series = (next[m.id] ?? []).filter((s) => s.t !== day);
    series.push(snap);
    next[m.id] = series;
  }
  return next;
}
